lib/MusicBrainz/Server/Data/Search.pm

```diff
@@ -1042,14 +1042,14 @@ sub xml_search
     $query = uri_escape_utf8($query);
 
     my $search_url_string;
-    if (DBDefs->SEARCH_ENGINE eq 'LUCENE') {
+    if ($version eq "1" or DBDefs->SEARCH_ENGINE eq 'LUCENE') {
         $search_url_string = "http://%s/ws/$version/%s/?query=%s&offset=%s&max=%s&fmt=xml";
     } else {
         $search_url_string = "http://%s/%s/edismax?q=%s&start=%s&rows=%s&wt=mbxml";
     }
 
     my $search_url = sprintf($search_url_string,
-                                 DBDefs->SEARCH_SERVER,
+                                 $version eq "1" ? DBDefs->OLD_SEARCH_SERVER : DBDefs->SEARCH_SERVER,
```

Comment:
Apparently the `xml_search` subroutine in here is *only* used by WS1, so the version check isn't necessary, but it's good for clarity. (WS2 uses the `xml_search` in `Data::WebService`. Not sure why they're separate.)